inspirehep/modules/disambiguation/core/ml/models.py

```diff
@@ -76,3 +102,161 @@ def fit(self):
 
     def predict(self, X):
         return self.estimator.predict(X)
+
+
+class DistanceEstimator(object):
+    def __init__(self, ethnicity_estimator_filename):
+        self.ethnicity_estimator = EthnicityEstimator()
+        self.ethnicity_estimator.load_model(ethnicity_estimator_filename)
+
+    def load_data(self, signatures_filename, records_filename, pairs_filename):
+        signatures, records = load_signatures(signatures_filename, records_filename)
+        pairs = json.load(open(pairs_filename, 'r'))
+
+        self.X = np.empty((len(pairs), 2), dtype=np.object)
+        self.y = np.empty(len(pairs), dtype=np.int)
+
+        for k, (i, j, target) in enumerate(pairs):
+            self.X[k, 0] = signatures[i]
+            self.X[k, 1] = signatures[j]
+            self.y[k] = target
```

Comment:
This is no longer correct, because each line of the `.jsonl` file now looks like this:
```json
{"same_cluster": True, "signature_uuids": ["cbf081db-fcb7-4386-baaf-f30636debfa7", "d08e1eb7-fa1b-4ea0-8917-5b3de969c582"]}
```